xarray/core/variable.py

```diff
@@ -761,7 +757,14 @@ def __getitem__(self, key) -> Self:
         array `x.values` directly.
         """
         dims, indexer, new_order = self._broadcast_indexes(key)
-        data = as_indexable(self._data)[indexer]
+        indexable = as_indexable(self._data)
```

Comment:
@shoyer we discussed this a while ago but would be good to get your perspective here again.

We are thinking of preserving the `*Indexer` objects in the returned values from `self._broadcast_indexes`, and then dispatching to `__getitem__`, `oindex` and `vindex` as appropriate on the underlying arrays. We will update all indexing adapters to provide all three methods.

Reply:
@shoyer, the gentlest of bumps on this. have a few minutes to take a look at this? :smile: